src/alire/alire-origins-deployers-system-portage.adb

```diff
@@ -0,0 +1,148 @@
+         --  Format: 0.0.0_release-rc9
+         if Tmp (C) = L1.Low_Line then
+            Tmp (C) := L1.Hyphen;
+
+            Trace.Debug ("To_Semantic_Version: " & Tmp);
+
+            return Tmp;
+         end if;
+      end loop;
+
+      Trace.Debug ("To_Semantic_Version: " & Tmp);
+
+      return Tmp;
+   end To_Semantic_Version;
+
+   ------------
+   -- Detect --
+   ------------
+
+   overriding
+   function Detect (This : Deployer) return Version_Outcomes.Outcome
+   is
+      function Split_Version (Gentoo_Package_Version : String) return String is
+      begin
+         for Index in Gentoo_Package_Version'Range loop
+            if Gentoo_Package_Version (Index) = L1.Hyphen then
+               return Gentoo_Package_Version
+                 (Index + 1 .. Gentoo_Package_Version'Last);
+            end if;
+         end loop;
+
+         return "";
+      end Split_Version;
+      pragma Unreferenced (Split_Version);
+
+      PortageQ_Output : constant AAA.Strings.Vector :=
+                    Subprocess.Checked_Spawn_And_Capture
+                      ("portageq",
+                       Empty_Vector & "best_visible" & "/" &
+                         This.Base.Package_Name,
+                       Valid_Exit_Codes => (0, 1), -- Returned when not found
+                       Err_To_Out       => True);
+      --  portageq should *always* produce zero or one line result.
+
+      Output    : constant AAA.Strings.Vector :=
+                    Subprocess.Checked_Spawn_And_Capture
+                      ("qatom",
+                       Empty_Vector & "-F%{PVR}" &
+                         First_Element (PortageQ_Output),
+                       Valid_Exit_Codes => (0, 1), -- Returned when not found
+                       Err_To_Out       => True);
+      Gentoo_Version : constant String := First_Element (Output);
+
+      --  Regexp : constant String :=
+      --    "[0-9]+(\.[0-9]+)*[a-z](_[a-z]+[0-9]?)*(-r[0-9]*)*";
+      --  From the pms.pdf.
+   begin
+      if Gentoo_Version /= "" then
+         Trace.Info ("Detect: " & This.Base.Package_Name & " - " &
+           Gentoo_Version & " - " & To_Semantic_Version (Gentoo_Version) &
+           " => " & Semantic_Versioning.Parse
+                         (To_Semantic_Version (Gentoo_Version),
+                            Relaxed => True).Image);
+
+         return
+            Version_Outcomes.New_Result
+               (Semantic_Versioning.Parse
+                 (To_Semantic_Version (Gentoo_Version), Relaxed => True));
+      end if;
+
+      Trace.Debug ("System deployer could not detect: " & This.Base.Image);
+      return Version_Outcomes.Outcome_Failure ("could not be detected",
+                                               Report => False);
+   end Detect;
+
+   -------------
+   -- Install --
+   -------------
+
+   overriding
+   function Install (This : Deployer) return Outcome is
+   begin
+      Trace.Info ("Install: " & This.Base.Package_Name);
```

Comment:
For consistency with other installers, this should be at most a Detail-level log.